### PR TITLE
Callout performance characteristics of ts-server

### PR DESCRIPTION
**Pull request by jdevries3133**
While reading the guide with an eye for improving the editor experience, I was not sure how ts-server behaves, or how fixing hotspots I view during tracing would change the editing experience. I ended up jumping into Discord for clarity, and this is a summary of what I learned.

### CODE REVIEW

File: Performance.md [hunk 1]
```diff
@@ -492,6 +492,16 @@ Isolated file emit can be leveraged by using the following tools:
...
 
+# Optimizing Editing Experience; Performance of `ts-server`
+
+`ts-server`'s performance characteristics will always be related to the
+performance of type-checking your entire project using `tsc`, so the other
+performance optimization guidance here also applies to improving the editing
+experience. As you type, the checker is completely starting from scratch, but it
+only requests information about what you're typing. This means that the editing
+experience can vary based on how much work TypeScript needs to do to check the
+type of what you are actively editing.
```
Comment (sandersn) [inline Performance.md:503]: I would like an additional sentence about errors:

1. the editor requests diagnostics in the background, a second or two after you stop typing.
2. the diagnostics are produced for all open files, not the entire project. So they will appear faster than a full `tsc` run, but slower than quick info, hover, etc. (As you mention already, error-reporting performance is closely related to `tsc` performance.)

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good, but I'd like a little more discussion of diagnostics-reporting performance of tsserver.

File: Performance.md (hunk 1 above)
Comment (DanielRosenwasser) [reply Performance.md:503]: (2) is true for VS Code I believe, but not for VS

so you could say "most editors like VS Code"

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
Can you reformat to one sentence per line?

File: Performance.md (hunk 1 above)
Comment (jdevries3133) [reply Performance.md]: OK, I added some details about diagnostics.

File: Performance.md [hunk 2]
```diff
@@ -492,6 +492,16 @@ Isolated file emit can be leveraged by using the following tools:
 * [ts-jest](https://kulshekhar.github.io/ts-jest/) can use be configured with the [`isolatedModules` flag set to `true`]isolatedModules: true(.
 * [ts-node](https://www.npmjs.com/package/ts-node) can detect [the `"transpileOnly"` option in the `"ts-node"` field of a `tsconfig.json`, and also has a `--transpile-only` flag](https://www.npmjs.com/package/ts-node#cli-and-programmatic-options).
 
+# Optimizing Editing Experience; Performance of `ts-server`
+
+`ts-server`'s performance characteristics will always be related to the
```
[the remark this answers is not in the document]
Comment (jdevries3133) [reply Performance.md]: Done

**Review by sandersn — APPROVED** [review]

**Pull request merged**